### PR TITLE
Allow plugins to register their metrics into Nrtsearch common collector registry

### PR DESCRIPTION
**Pull request by vim345**

### CODE REVIEW

File: src/main/java/com/yelp/nrtsearch/server/plugins/MetricsPlugin.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import io.prometheus.client.CollectorRegistry;
+
+/**
+ * Plugin interface that allows plugin to register their prometheus metrics in Nrtsearch prometheus
+ * collector registry.
+ */
+public interface MetricsPlugin {
+
+  /** @param collectorRegistry Nrtsearch Prometheus collector registry. */
+  void registerCollectorRegistry(CollectorRegistry collectorRegistry);
```
Comment (sarthakn7) [inline src/main/java/com/yelp/nrtsearch/server/plugins/MetricsPlugin.java:27]: Nit: rename this to `registerMetrics`

Comment (vim345) [reply src/main/java/com/yelp/nrtsearch/server/plugins/MetricsPlugin.java]: Done.

File: src/test/java/com/yelp/nrtsearch/server/plugins/PluginsServiceTest.java [hunk 2]
```diff
@@ -122,22 +127,28 @@ public void testGetPluginJars() throws IOException {
...
     folder.newFile("some_file.txt");
     folder.newFolder("some_folder.jar");
-    PluginsService pluginsService = new PluginsService(getEmptyConfig());
+    PluginsService pluginsService = new PluginsService(getEmptyConfig(), getCollectorRegistry());
     assertEquals(
         new HashSet<>(jars), new HashSet<>(pluginsService.getPluginJars(folder.getRoot())));
   }
 
-  public static class LoadTestPlugin extends Plugin {
+  public static class LoadTestPlugin extends Plugin implements MetricsPlugin {
```
Comment (aprudhomme) [inline src/test/java/com/yelp/nrtsearch/server/plugins/PluginsServiceTest.java:135]: Maybe instead making this class a `MetricsPlugin`, create a second class and add a separate test to verify it gets the registry and this one doesn't.

File: src/main/java/com/yelp/nrtsearch/server/plugins/PluginsService.java [hunk 3]
```diff
@@ -176,9 +179,14 @@ Class<? extends Plugin> getPluginClass(List<File> jarList, String pluginClassNam
   /** Get an instance of the loaded plugin class using reflection. */
   Plugin getPluginInstance(Class<? extends Plugin> pluginClass) {
     try {
-      return pluginClass
-          .getDeclaredConstructor(new Class[] {LuceneServerConfiguration.class})
-          .newInstance(config);
+      Plugin pClass =
```
Comment (aprudhomme) [inline src/main/java/com/yelp/nrtsearch/server/plugins/PluginsService.java:182]: nit: this variable is a plugin instance, not a class. Perhaps there is a more appropriate name

Comment (vim345) [reply src/main/java/com/yelp/nrtsearch/server/plugins/PluginsService.java]: Godo point.

File: src/test/java/com/yelp/nrtsearch/server/plugins/PluginsServiceTest.java (hunk 2 above)
Comment (vim345) [reply src/test/java/com/yelp/nrtsearch/server/plugins/PluginsServiceTest.java]: Done.

**Review by aprudhomme — APPROVED** [review]
Looks ok to me

**Pull request merged**